src/components/filters/FilterSection/index.jsx

```diff
@@ -297,37 +326,71 @@ class FilterSection extends React.Component {
           {
-            this.state.isExpanded
+            isSearchFilter && this.state.isExpanded
+              ? this.props.options
+                // .filter(option => this.state.optionsVisibleStatus[option.text])
+                .map(option => (
+                  // We use the 'key' prop to force the SingleSelectFilter
+                  // to rerender on filterStatus change.
+                  // See https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key
+                  <SingleSelectFilter
+                    key={`${option.text}-${filterStatus[option.text] ? 'enabled' : 'disabled'}`}
+                    label={option.text}
+                    onSelect={label => this.handleSelectSingleSelectFilter(label)}
+                    selected={filterStatus[option.text]}
+                    count={null}
+                    hideZero={false}
+                    accessible={option.accessible}
+                    tierAccessLimit={this.props.tierAccessLimit}
+                    disabled={option.disabled}
+                    lockedTooltipMessage={this.props.lockedTooltipMessage}
+                    disabledTooltipMessage={this.props.disabledTooltipMessage}
+                  />
+                )) : null
+          }
+          {
+            (isTextFilter && this.state.isExpanded)
               ? this.props.options
                 .filter(option => this.state.optionsVisibleStatus[option.text])
                 .map((option, index) => {
                   if (index >= this.props.initVisibleItemNumber && !this.state.showingMore) {
                     return null;
                   }
-                  if (option.filterType === 'singleSelect') {
-                    return (
-                      // We use the 'key' prop to force the SingleSelectFilter
-                      // to rerender on filterStatus change.
-                      // See https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key
-                      <SingleSelectFilter
-                        key={`${option.text}-${filterStatus[option.text] ? 'enabled' : 'disabled'}`}
-                        label={option.text}
-                        onSelect={label => this.handleSelectSingleSelectFilter(label)}
-                        selected={filterStatus[option.text]}
-                        count={option.count}
-                        hideZero={this.props.hideZero}
-                        accessible={option.accessible}
-                        tierAccessLimit={this.props.tierAccessLimit}
-                        disabled={option.disabled}
-                        lockedTooltipMessage={this.props.lockedTooltipMessage}
-                        disabledTooltipMessage={this.props.disabledTooltipMessage}
-                      />
-                    );
+                  return (
```

Comment:
feels like this part is a bit redundant comparing to https://github.com/uc-cdis/gen3-ui-component/pull/78/files#diff-07a74b594d96598ad424ba2c977d9a88R337-R357

maybe we can filter the `options` according to the controlling variables before?

Reply:
Gotcha, let me try to un-duplicate this code -- I'll see if I can do it in a way that isn't too confusing.